### PR TITLE
add category and quiz page

### PR DESCRIPTION
**Pull request by iRohitGaur**
[Files to check](https://github.com/iRohitGaur/StarTrivia/pull/4/files?file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&show-viewed-files=true)

[Preview](https://deploy-preview-4--startrivia.netlify.app)

### CODE REVIEW

File: src/pages/quiz/Quiz.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                        navigate("/result", { quiz, score });
+                      } else {
+                        setQuestion((q) => q + 1);
+                      }
+                      setAnswered(initialAns);
+                    }}
+                    className="sui_btn btn_v2 flex_row flex_align_center flex_gap1"
+                  >
+                    {question + 1 === quiz.mcqs.length ? "Finish" : "Next"}
+                  </button>
```
Comment (b0llu) [inline src/pages/quiz/Quiz.jsx:53]: Liked the way how you are handling the end of quiz

File: src/reducers/categorySlice.js [hunk 2]
```diff
@@ -35,6 +61,28 @@ const categorySlice = createSlice({
...
+      .addCase(getQuiz.pending, (state) => {
+        state.loading = true;
+      })
+      .addCase(getQuiz.fulfilled, (state, action) => {
+        state.loading = false;
+        state.quiz = action.payload;
+      })
+      .addCase(getQuiz.rejected, (state, action) => {
+        state.loading = false;
+        toast.error(action.payload);
```
Comment (b0llu) [inline src/reducers/categorySlice.js:85]: its great that you are using the .rejected for toast errors

**Review by b0llu — APPROVED** [review]
Everything looks good. Code is clean you can merge this

File: src/reducers/categorySlice.js [hunk 3]
```diff
@@ -35,6 +61,28 @@ const categorySlice = createSlice({
...
+      })
+      .addCase(getQuizesInCategory.pending, (state) => {
+        state.loading = true;
+      })
+      .addCase(getQuizesInCategory.fulfilled, (state, action) => {
+        state.loading = false;
+        state.quizes = action.payload;
+      })
+      .addCase(getQuizesInCategory.rejected, (state, action) => {
+        state.loading = false;
```
Comment (manojsarna) [inline src/reducers/categorySlice.js:73]: Appreciation : adding the cases for rejected and pending is good

File: src/pages/category/Category.jsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <div className="category_page_wrapper">
+      {categories.length !== 0 && quizes.length !== 0 && (
+        <div className="landing_wrapper flex_column flex_gap2">
+          <div className="title_wrapper flex_column flex_align_center flex_gap2">
+            <div className="back_title_wrapper flex_row flex_align_center flex_justify_center">
+              <div className="category_cta flex_row flex_align_center flex_gap2">
+                <button
+                  className="sui_btn btn_v1 flex_row flex_align_center flex_gap1"
+                  onClick={() => navigate("/")}
```
Comment (itsmebipul00) [inline src/pages/category/Category.jsx:35]: Suggestion: navigate(-1) goes back ??

**Review by bhushannaikaroskar — APPROVED** [review]
Good job on UI and code is clean. good to merge

File: src/reducers/categorySlice.js (hunk 2 above)
Comment (itsmebipul00) [reply src/reducers/categorySlice.js:85]: Really nicely done

File: src/pages/category/Category.jsx (hunk 4 above)
Comment (iRohitGaur) [reply src/pages/category/Category.jsx:35]: yes it does, but it will not work for the case when you open the category page directly from a link.

**Review by itsmebipul00 — APPROVED** [review]
UI is awesome Code is amazing !! Good to merge
Ohh and about the navigate(-1), after previewing your UI I noticed it

File: src/pages/category/Category.jsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      )}
+      {loading && (
+        <div className="stg_loader">
+          <Loader
+            height="150"
+            width="150"
+            color="#1a8d1a"
+            ariaLabel="loading"
+          />
+        </div>
```
Comment (manojsarna) [inline src/pages/category/Category.jsx:67]: Appreciation : good use of react package for loaders

**Review by manojsarna — APPROVED** [review]
Overall the code is good

UI Review : UI is clean and looking really nice :)

**Pull request merged**